baybe/recommenders/pure/bayesian/botorch.py

```diff
@@ -47,14 +48,24 @@ class BotorchRecommender(BayesianRecommender):
     # See base class.
 
     # Object variables
+    num_restarts: int = field(validator=[instance_of(int), gt(0)], default=10)
```

Comment:
Hey @Scienfitz, thanks for the proposal. I have a few comments we should discuss. Which of them apply in the depends on the final layout we use, but here just listed one by one:
* In the current layout, the names break our naming convention. That is, if we expose them directly as attributes of our own class, I think we should follow **our** convention, not the botorch one, i.e. use `n_...` to indicate that this is a number.
* Perhaps you didn't notice, but you actually introduced a breaking change here by placing the new ones *before* the existing ones (i.e. incompatible when things are passed as positional arguments). However, I think it perhaps makes sense to change all "behavior-controlling" attributes into actual kw-only arguments.
* Maybe one other thought, to prevent potential future problems: `optimize_acqf` supports many more arguments and we've more or now picked two "random" ones that we explicitly expose. Chances are that more will follow and I'm not sure if it's the best idea to just continue mirroring everything to our classes, what do you think? I see two potential alternatives: 1) expose an explicit `botorch_kwargs` dictionary argument (or named similarly) that collects everything in one place or even 2) just allow `**kwargs` in the constructor. In both cases, the question is if we want to convert/validate this stuff as well or not. Optional but would be possible with an internal options helper class to which stuff is converted first. 

Let me know your thoughts.

Reply:
1) Ok, can change the names. You often refer to `our naming scheme`, and while such a thing might exist implicitly, did we ever agree or note that down?

2) Right I didnt think about the breaking, but thought about the kwargs too. So I would definitely add those, but it would remain kind of breaking right?

3) Not sure thats necessary, if the constructor allows overall unvalidated `kwargs` I'd be down for it - perhaps thats a deign patternw e can allow for more classes that are very close to botorch functions. But overall I dont see much reason for changing the approach of adding the important flags once they become relevant somewhere (like here I was doing a use case and noticed strong inconsistency between the recommendations and wanted to increase these numbers). This has one major advantage: we can take the opportunity and properly validate and document these flags so everyone can in principle understand them without leaving baybe. If you leave that to people looking up Botorch... you know how well one finds things there

Reply:
1. True, we never made it explicit or agreed to anything. And I'm open to do so, if you have suggestions. My only requirements are consistency and clarity – both of which I feel is often lacking in botorch. For example, why do they call it `num_restarts` but not `num_raw_samples` instead of `raw_samples`, given that both are integers? In fact, their naming rather implies to me that `raw_samples` expects the actual samples, not their quantity. You know what I mean... So any inputs welcome
2. Adding `kw_only` to existing arguments --> yes, would be breaking. So your choice how you want to do it. We can either i) add the `kw_only` only to the new arguments and later in the major release we also add them to the existing ones as a breaking change or ii) we add it now everywhere and accept a breaking change in the minor version (in both cases we should mention it in the changelog) or iii) we leave the `kw_only` for now and you add the new attributes to the end of the list
3. I think it all boils down to whether we want documentation/validation or not. So what is your preference? Asking because your statements are slightly contradictory (first you say unvalidated kwargs would be OK but later you argue pro validated/documented args). BTW: unvalidated kwargs are possible 👍🏼 However, it's also OK for me if we add the current ones as explicit args for now and adjust later if needed.

Reply:
1. Names changed accordingly
2. Attributes moved to the end, `kw_only` not touched
3. My comment should be read more like: I don't see need for change and in particular the ability to explain and doc is something useful in the current approach. But IF it was changed, the undocumented fully flexible kwargs would be my choice.

Reply:
Kay, thanks for clarification. I think we're good for this PR